gdal/gcore/gdaldataset.cpp

```diff
@@ -3344,7 +3344,12 @@ char ** GDALDataset::GetMetadata(const char * pszDomain)
 char ** GDALDataset::GetMetadataDomainList()
 {
   char ** currentDomainList = CSLDuplicate(oMDMD.GetDomainList());
-  currentDomainList = CSLAddString(currentDomainList,"DERIVED_SUBDATASETS");
+
+  // Ensure that we do not duplicate DERIVED domain
+  if(CSLFindString(currentDomainList,"DERIVED_SUBDATASETS")==-1)
```

Comment:
In some of the tests from jp2openjpeg.py (the ones with vsimem). It seems that the dataset is a duplicate from another dataset (and that the whole metadata domain list is already ported during duplication).